example/server.js

```diff
@@ -2,22 +2,51 @@
 
 var Hapi = require('hapi');
 var Hoek = require('hoek');
-var tags = require('../lib/index.js');
+var fs = require('fs');
+var pg = require('pg');
 
 var server = new Hapi.Server();
+var tagsData = JSON.parse(fs.readFileSync('./example/tags.json').toString());
+var tags = require('../lib/index.js');
+var tagsPool = new pg.Pool();
+
+
+function init (port, callback) {
+  server.connection({ port: port });
+
+  server.register([{
+    register: tags,
+    options: { tags: tagsData, tagsPool: tagsPool }
+  }], function (err) {
+    if (err) {
+      return callback(err);
+    }
+
+    server.route([{
+      method: 'GET',
+      path: '/',
+      handler: function (request, reply) {
+        tagsPool.connect(function (connErr, client, done) {
+          Hoek.assert(!connErr, connErr);
+          client.query('select * from tags', function (dberr, res) {
+            Hoek.assert(!dberr, dberr);
+            done();
 
-server.connection({ port: process.env.PORT || 3000 });
+            return reply(res.rows);
+          });
+        });
+      }
+    }, {
+      method: 'GET',
+      path: '/hello',
+      handler: function (request, reply) {
+        reply('hello');
+      }
+    }]);
 
-server.register(tags, function (err) {
-  Hoek.assert(!err, 'error registering plugin');
-});
+    return callback(null, server);
+  });
+}
 
-server.route({
-  method: 'GET',
-  path: '/',
-  handler: function (request, reply) {
-    reply(request.jackmisawesome);
-  }
-});
 
-module.exports = server;
+module.exports = { init: init, tagsPool: tagsPool };
```

Comment:
Ok, I can see how this is beneficial for this module in that it keeps the example server as simple as possible. But if/when we move to our app I think it would be useful to have access to the pool  in the app.

If we start the pool and expose it with the server, then we will have to change the way that we test that the DB has been initialised and that it is empty, won't we? At the moment I have a test, which uses the pool before initialising the server.